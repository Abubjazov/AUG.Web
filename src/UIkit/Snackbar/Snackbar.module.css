.root {
  width: 100%;
  padding: 10px;
  color: var(--white);
  border: 1px solid rgb(227 220 255 / 40%);
  border-radius: 10px;
  box-shadow: var(--box-shadow);
  transition: all var(--transition-delay) linear;
  animation: var(--animation-visible);
}

.error {
  background: var(--red);
}

.info {
  background: var(--blue);
}

.title {
  font-size: 16px;
  font-weight: 400;
  line-height: 20px;
  text-align: center;
}
