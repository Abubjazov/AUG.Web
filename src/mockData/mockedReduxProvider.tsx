import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockMyDapplets,
  mockMyTags,
} from './mockData'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import myDappletsReducer from '../store/slices/myDappletsSlice'

export const defaultMockState = {
  layout: {
    menuOpened: false,
    dappletSettingsOpened: false,
  },

  myDapplets: {
    myDapplets: [...mockMyDapplets],
    myTags: [...mockMyTags],
  },

  dapplets: {
    dapplets: [...mockDapplets],
    tags: [...mockCommunityTags],
  },
}

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  const store = configureStore({
    reducer: {
      layout: layoutReducer,
      dapplets: dappletsReducer,
      myDapplets: myDappletsReducer,
    },
    preloadedState: {
      ...defaultMockState,
    },
  })

  return <Provider store={store}>{children}</Provider>
}
