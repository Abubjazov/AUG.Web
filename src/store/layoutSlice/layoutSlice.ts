import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import type { RootState } from '../index'

type TLayout = { menuOpened: boolean; dappletSettingsOpened: boolean }

const windowInnerWidth = window.innerWidth

export const getInitialState = (windowInnerWidth: number, setPoint: number) =>
  windowInnerWidth > setPoint ? true : false

const initialState: TLayout = {
  menuOpened: getInitialState(windowInnerWidth, 1300),
  dappletSettingsOpened: getInitialState(windowInnerWidth, 1600),
}

export const layoutSlice = createSlice({
  name: 'layout',
  initialState,
  reducers: {
    setDappletSettingsState: (state, action: PayloadAction<boolean>) => {
      state.dappletSettingsOpened = action.payload
    },

    setMenuState: (state, action: PayloadAction<boolean>) => {
      state.menuOpened = action.payload
    },
  },
})

export const { setDappletSettingsState, setMenuState } = layoutSlice.actions

export const selectLayout = (state: RootState) => state.layout

export default layoutSlice.reducer
