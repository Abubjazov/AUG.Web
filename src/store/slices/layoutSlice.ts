import { ReactNode } from 'react'

import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import { getInitialState } from 'utils/getInItialState/getInItialState'

import type { RootState } from '../index'

type TLayout = {
  menuOpened: boolean
  dappletSettingsOpened: boolean
  modalState: boolean
  modalContent: ReactNode
}

const windowInnerWidth = window.innerWidth

const initialState: TLayout = {
  menuOpened: getInitialState(windowInnerWidth, 1300),
  dappletSettingsOpened: getInitialState(windowInnerWidth, 1600),
  modalState: true,
  modalContent: null,
}

export const layoutSlice = createSlice({
  name: 'layout',
  initialState,
  reducers: {
    setDappletSettingsState: (state, action: PayloadAction<boolean>) => {
      state.dappletSettingsOpened = action.payload
    },

    setMenuState: (state, action: PayloadAction<boolean>) => {
      state.menuOpened = action.payload
    },

    setModalState: (state, action: PayloadAction<boolean>) => {
      state.modalState = action.payload
    },

    setModalContent: (state, action: PayloadAction<ReactNode>) => {
      state.modalContent = action.payload
    },
  },
})

export const {
  setDappletSettingsState,
  setMenuState,
  setModalState,
  setModalContent,
} = layoutSlice.actions

export const selectLayout = (state: RootState) => state.layout

export default layoutSlice.reducer
