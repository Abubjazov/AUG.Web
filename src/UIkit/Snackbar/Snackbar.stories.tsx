import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider from 'mockData/mockedReduxProvider'
import { EMessageType } from 'store/slices/layoutSlice'

import Snackbar, { SnackbarProps } from './Snackbar'

import '/src/index.css'

const meta: Meta<SnackbarProps> = {
  component: Snackbar,
  title: 'UIkit/Snackbar',
  parameters: {
    docs: {
      description: {
        component:
          'Interface component - Snackbar. Component for visualizing service message or error.',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    showMode: {
      description:
        'Activate show mode for StoryBook (deactivates self-destruction of message from the message queue)',
    },
    message: {
      description: 'Message/Error data object (IMessage)',
    },
  },
}

export default meta

type Story = StoryObj<SnackbarProps>

export const Default: Story = {
  args: {
    showMode: true,
    message: {
      messageId: '31',
      messageText: 'INFO Message text',
      messageType: EMessageType.INFO,
    },
  },
  decorators: [
    Story => (
      <MockedProvider>
        <div
          style={{
            maxWidth: '300px',
          }}
        >
          <Story />
        </div>
      </MockedProvider>
    ),
  ],
}
