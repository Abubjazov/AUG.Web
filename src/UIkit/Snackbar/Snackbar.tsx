import { FC, useEffect } from 'react'

import { useAppDispatch } from 'store/hooks'
import { IMessage, removeMessage } from 'store/slices/layoutSlice'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './Snackbar.module.css'

export interface SnackbarProps {
  userStyles?: string
  message: IMessage
}

const Snackbar: FC<SnackbarProps> = ({ userStyles = '', message }) => {
  const dispatch = useAppDispatch()

  const { messageId, messageText, messageType } = message

  const closeSnackBar = () => {
    dispatch(removeMessage(messageId))
  }

  useEffect(() => {
    const removeSnackBar = setTimeout(
      () => dispatch(removeMessage(messageId)),
      3500,
    )

    return () => clearTimeout(removeSnackBar)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <div
      data-testid="snackbar"
      className={cc([styles.root, styles[messageType], userStyles])}
      onClick={closeSnackBar}
    >
      <span className={styles['title']}>{messageText}</span>
    </div>
  )
}

export default Snackbar
