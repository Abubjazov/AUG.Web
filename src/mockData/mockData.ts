import { IDapplet, ITag } from 'store/slices/dappletsSlice'
import { IList, IUserDapplet } from 'store/slices/userDataSlice'

export const mockUserTags: ITag[] = [
  { tagId: 'darP5Jyz8yirTMsfY9RP', tagName: 'Twitter' },
  { tagId: 'JW3UFtZ5HgATcwldsJ1T', tagName: 'Facebook' },
  { tagId: '6UKxNrMzte6RCWIlssvM', tagName: 'Top 10' },
  { tagId: 'ZbEuko9nOlQYKG6I1Efx', tagName: 'Test' },
  { tagId: '521keF1Adymn8Oo896vv', tagName: 'ToDo' },
]

export const mockUserLists: IList[] = [
  { listId: 'XfLdXIykGHsgIjdJkTaE', listName: 'My special list' },
  { listId: 'c6rClgOltChco2XnyVDp', listName: 'Financial list' },
]

export const mockUserDapplets: IUserDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    userTags: ['JW3UFtZ5HgATcwldsJ1T', '521keF1Adymn8Oo896vv'],
    dappletState: true,
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    userTags: [],
    dappletState: true,
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    userTags: 'N/A',
    dappletState: true,
  },
]

export const mockCommunityTags: ITag[] = [
  { tagId: 'XoEVTHXhfMAjIWsi4wxY', tagName: 'Top 100' },
  { tagId: '3LYUFQTbdNULwBgCkfXx', tagName: 'Finances' },
  { tagId: 'XZ2zPZqa4fPxM9mLA9xR', tagName: 'Media' },
  { tagId: 'C69B5fQ7ccW6Q1SC0yow', tagName: 'Favourites' },
]

export const mockDapplets: IDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    appOwner: 'Satoshi Nakamoto',
    circulatingSupply: '19534212',
    communityTags: ['3LYUFQTbdNULwBgCkfXx', 'C69B5fQ7ccW6Q1SC0yow'],
    date: '1230757200',
    fullDesc:
      'Bitcoin is a peer-to-peer online currency, meaning that all transactions happen directly between equal, independent network participants, without the need for any intermediary to permit or facilitate them. Bitcoin was created, according to Nakamoto’s own words, to allow “online payments to be sent directly from one party to another without going through a financial institution.”',
    fullyDilutedMarketCap: '738866583732',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-1.01',
    maxSupply: '21000000',
    name: 'Bitcoin Contracts',
    shortDesc:
      'Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. ',
    shortName: 'BTC',
    totalSupply: '19534106',
    volume: '10.9',
    volumePerMarketCap: '1.58',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHGCOPY',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    appOwner: 'XRP Ledger',
    circulatingSupply: '53615837759',
    communityTags: 'N/A',
    date: '1620939600',
    fullDesc:
      'Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.',
    fullyDilutedMarketCap: '65183420834',
    logo: '/public/images/notAvailable.svg',
    marketCap: '5.66',
    maxSupply: '100000000000',
    name: 'XRP Ledger',
    shortDesc:
      'Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.',
    shortName: 'XRP',
    totalSupply: '99988316618',
    volume: '132.26',
    volumePerMarketCap: '5.37',
  },
  {
    dappletId: 'QbWG3sKvfgFcP5RtskMp',
    appOwner: 'Realcoin',
    circulatingSupply: '85367759449',
    communityTags: [],
    date: '1388523600',
    fullDesc:
      'Launched in 2014, Tether is a blockchain-enabled platform designed to facilitate the use of fiat currencies in a digital manner. Tether works to disrupt the conventional financial system via a more modern approach to money. Tether has made headway by giving customers the ability to transact with traditional currencies across the blockchain, without the inherent volatility and complexity typically associated with a digital currency. As the first blockchain-enabled platform to facilitate the digital use of traditional currencies (a familiar, stable accounting unit), Tether has democratised cross-border transactions across the blockchain.',
    fullyDilutedMarketCap: '88708606718',
    logo: '/dappletsLogos/tether.png',
    marketCap: '0.01',
    maxSupply: 'N/A',
    name: 'Tether USDt',
    shortDesc:
      'Tether works to disrupt the conventional financial system via a more modern approach to money.',
    shortName: 'USDT',
    totalSupply: '88623656724',
    volume: '41.44',
    volumePerMarketCap: '32.16',
  },
]
