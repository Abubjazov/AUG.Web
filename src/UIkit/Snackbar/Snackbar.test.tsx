import { render } from '@testing-library/react'
import { mockedReduxProvider as Provider } from 'mockData/mockedReduxProvider'
import * as actions from 'store/slices/layoutSlice'
import { EMessageType } from 'store/slices/layoutSlice'

import Snackbar from './Snackbar'

function delay(timeout: number | undefined) {
  return new Promise((resolve: (value: unknown) => void) => {
    setTimeout(resolve, timeout)
  })
}

describe('Snackbar', () => {
  test('should render Snackbar "info"', () => {
    const { asFragment } = render(
      <Provider>
        <Snackbar
          message={{
            messageId: '13',
            messageText: 'Message text',
            messageType: EMessageType.INFO,
          }}
        />
        ,
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render Snackbar "error"', () => {
    const { asFragment } = render(
      <Provider>
        <Snackbar
          message={{
            messageId: '13',
            messageText: 'Message text',
            messageType: EMessageType.ERROR,
          }}
        />
        ,
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call removeMessage function on input text', async () => {
    const mockedRemoveMessage = vi.spyOn(actions, 'removeMessage')

    render(
      <Provider>
        <Snackbar
          message={{
            messageId: '13',
            messageText: 'Message text',
            messageType: EMessageType.ERROR,
          }}
        />
      </Provider>,
    )

    await delay(3500)

    expect(mockedRemoveMessage).toHaveBeenCalledTimes(1)
    expect(mockedRemoveMessage).toHaveBeenCalledWith('13')
  })
})
