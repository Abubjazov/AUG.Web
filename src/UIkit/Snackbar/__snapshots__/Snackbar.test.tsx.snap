// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snackbar > should render Snackbar "error" 1`] = `
<DocumentFragment>
  <div
    class="_root_4e3dca _error_4e3dca "
    data-testid="snackbar"
  >
    <span
      class="_title_4e3dca"
    >
      Message text
    </span>
  </div>
  ,
</DocumentFragment>
`;

exports[`Snackbar > should render Snackbar "info" 1`] = `
<DocumentFragment>
  <div
    class="_root_4e3dca _info_4e3dca "
    data-testid="snackbar"
  >
    <span
      class="_title_4e3dca"
    >
      Message text
    </span>
  </div>
  ,
</DocumentFragment>
`;
